doc/iterations/2024.3/shapes/add_support_for_browse_upload_images_widget.adoc

```diff
@@ -0,0 +1,95 @@
+* _Is Enabled Expression_
+
+* _AddImageOperation_ with a _body_ to specify the action to apply once an image is selected.
+* _RemoveImageOperation_ with a _body_ to specify the action to apply once an image is deleted.
+* _style_ (with its corresponding _conditionalStyles_) to configure the label
+
+This new widget will be based on the UI of the reference widget, using the MUI autocomplete component.
+
+* The selected images will be displayed in _chips_ with preview and label.
+It must be possible to delete a selected image directly from its _chips_.
+* The options will be populated by images marked as favourite.
+* A trash can icon will delete the current selection.
+* The image selection dialog will open using `...` button.
+
+image:images/widget_images/widget_images_chips.png[Widget-images chips example]
+
+=== Image selection dialog
+
+This dialog will have three menus: _Gallery_, _Link_ and _Import_.
+The default open menu is _Gallery_.
+
+A _Select_ button is always displayed at the bottom right of the dialog.
+
+Upon selecting, the image will be uploaded on the server (case _import_ and _link_), and the widget body must be called with the selected image.
+
+image:images/widget_images/widget_images_dialog.png[Image selection dialog]
+
+==== Gallery
+
+The gallery shows a thumbnail of the images already on the server.
+For each image, displays the preview, the label and a "like" button to add or remove the image to the favorite ones.
+
+* There is a filter bar to search image everywhere in the gallery from the image label.
+* There are three sub-menus to organize the gallery :
+** `Recents` only show the last added images
+** `Favorites` only show the "liked" images
+** `All` show all the imagesC
+
+_Select_ button must be enabled only if an image is selected.
+
+==== Link
+
+The link menu shows a textfield to type the image url.
+
+image:images/widget_images/widget_images_link_1.png[Image link, url selection]
+
+* If the url is valid, add a textfield to type the image label and the image preview.
+
+image:images/widget_images/widget_images_link_2.png[Image link, valid URL]
+
+* If the url is not a valid image, show an error message.
+
+image:images/widget_images/widget_images_link_3.png[Image link, invalid URL]
+
+_Select_ button must be enabled only if the url is a valid image and the label is not empty.
+
+==== Import
+
+The import menu shows a textfield to type the image label and a _select file_ area.
+_Select_ button must be enabled only if a document is uploaded, and the label is not empty.
+
+image:images/widget_images/widget_images_import.png[Image import]
+
+== Rabbit holes
+
+* There is no UI to remove an image from the server.
+* An example of this widget could be added in sirius-web to `NodeStylePropertiesConfigurer` by replacing the existing select and image widget for `ImageNodeStyle` properties page.
+
+== Cutting backs
+
+* All the favourite's support can be removed if deadlines are not respected.
+
+== No-gos
+
```

Comment:
What about the existing `/projects/projectId/settings` page allowing to upload images? Will this page be updated to use the new widget?

Reply:
I didn't know about this feature, I wonder if we can use the new widget directly.

This page only allows you to upload new images that will have a scope linked to the project (which is not possible with the widget image in its current version). I don't think it would be easy to apply an AddImageOperation to it.

Reply:
Ok

Reply:
I'm not sure I understand this part:

> This page only allows you to upload new images that will have a scope linked to the project which is not possible with the widget image in its current version

What does "the widget image in its current version" is? This current ADR? Because the scope and lifecycle of the images will have to be determined.